src/verification/android.rs

```diff
@@ -280,8 +265,11 @@ impl ServerCertVerifier for Verifier {
     ) -> Result<rustls::client::ServerCertVerified, TlsError> {
         log_server_cert(end_entity);
 
-        let server = match server_name {
-            rustls::ServerName::DnsName(name) => name.as_ref(),
+        // Verify the server name is one that we support and extract a string to use
+        // for the platform verifier call.
+        let server_name_str = match server_name {
+            ServerName::DnsName(dns_name) => dns_name.as_ref().to_string(),
+            ServerName::IpAddress(ip_addr) => ip_addr.to_string(),
             _ => return Err(unsupported_server_name()),
         };
```

Comment:
Suggestion: We could use a late binding here to avoid the extra hot-path allocation. We did the same thing in the the Apple and Windows verifier as well: https://github.com/rustls/rustls-platform-verifier/blob/1a5250c61d30fbac4af26c4de07561755359b4f6/src/verification/apple.rs#L194-L202

https://github.com/rustls/rustls-platform-verifier/blob/1a5250c61d30fbac4af26c4de07561755359b4f6/src/verification/windows.rs#L608-L614

Reply:
>  We could use a late binding here

Ohhhh! So clever. I'm not sure I realized this was possible :sweat_smile: It neatly resolves the issue I had where I couldn't return a ref to an intermediate var scoped within the match.

Thanks for the suggestion :+1: